### PR TITLE
Describe the risks of filtering at the DS

### LINKED ISSUE #251: Can the Delivery Service actually filter Commit messages?
The document says (5.2.1)

> The Delivery Service can rely on the epoch and content_type fields of an MLSMessage for providing an order only to handshake messages, and possibly even filter or reject redundant Commit messages proactively to prevent them from being broadcast

I think this may be a problem if a malformed Commit for epoch `n` is sent by one of the group members, and this Commit is selected by the Delivery Service:
- group members will reject this Commit and stay at epoch `n`
- the Delivery Service will think the group has moved to epoch `n+1` and reject further Commits that may be sent by other group members (that are still in epoch `n`)

### PR DESCRIPTION
**Pull request by bifurcation**
Fixes #251

### CODE REVIEW

File: draft-ietf-mls-architecture.md [hunk 1]
```diff
@@ -763,6 +763,28 @@ filter or reject redundant Commit messages proactively to prevent them from
...
+epoch is `n`, and as a result, the group is stuck -- no member can send a Commit
+that the DS will accept.
+
+Given these risks, it is effectively impossible for a strongly consistent DS to
+know with absolute certainty when it is safe to update its internal state.
+Nonetheless, there are some simple mechanisms that can mitigate these risks.
+For example, having the DS wait to update its state until it gets confirmation
+of acceptance from a quorum of members provides some protection against buggy
+clients.  It is up to the designers and operators of a DS to ensure that
+sufficient mechanisms are in place to address these risks.
```
Comment (Bren2010) [inline draft-ietf-mls-architecture.md:786]: I believe this PR is essentially the same as mine except that you've removed the proposal to defer to clients / allow forks, with a proposal to have a quorum. Is that a correct statement? Why do you think quorum is better than allowing forks, when quorum requires all group members to be online and forking can be offline?

**Comment by TWal** [comment]
I think this PR does a good job at framing the problem described in #251, however I think it would be best to refrain from trying to give solutions: the only solution would be to transmit all commits in a consistent order, and any attempt to filter may lead to DoS. The architecture document should allow the DS to filter (because actual deployments will filter), but warn about the risks it implies.

**Comment by ekr** [comment]
I propose we take Richard's first two grafs more or less as-is and then reduce the last graf to say:

> Given these risks, it is effectively impossible for a strongly consistent DS to
know with absolute certainty when it is safe to update its internal state.
It is up to the designers and operators of a DS to ensure that
sufficient mechanisms are in place to address these risks.

**Comment by bifurcation** [comment]
@ekr done

**Review by ekr — APPROVED** [review]
LGTM. @rohanmahy  any objections?

**Review by rohanmahy — APPROVED** [review]
works for me

**Pull request merged**